nupic/research/frameworks/pytorch/model_utils.py

```diff
@@ -91,16 +91,17 @@ def train_model(
                 "Mixed precision requires NVIDA APEX."
                 "Please install apex from https://www.github.com/nvidia/apex")
 
+    t0 = time.time()
     for batch_idx, (data, target) in enumerate(loader):
         num_images = len(target)
-        t1 = time.time()
         if batch_idx >= batches_in_epoch:
             break
         if pre_batch_callback is not None:
             pre_batch_callback(model=model, batch_idx=batch_idx)
 
         data = data.to(device, non_blocking=async_gpu)
         target = target.to(device, non_blocking=async_gpu)
+        t1 = time.time()
```

Comment:
You are combining "data loading" time with "pre_batch_callback" time and "cpu/gpu" memory transfer time with "t1"

Reply:
Yes that's true, but currently that callback doesn't do anything. I wanted to include the .to() call but that may not do much - should I just move t1 to be earlier instead?

Reply:
> should I just move t1 to be earlier instead?

It depends what is being profiled. 
I was just pointing out that if you include the ".to()" timing to "t1" then "t1" represents "loading data from disk/memory" and  "moving data from cpu to gpu". If you assume "moving data to gpu" does not impact your stats then it is ok. 
When i used pytorch profiler, i found out that most of the time was spent on moving data from cpu to gpu via ".item()", that's why "validate()" is still slow.